### PR TITLE
Spring boot 2.x

### PR DESCRIPTION
**Pull request by Blackbaud-MikeLueders**
@blackbaud/java-internal-tooling

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -16,16 +19,15 @@ dependencies {
...
-    compile "org.gradle:gradle-tooling-api:${gradleVersion}"
 
-    testCompile gradleTestKit()
-    testCompile ('com.blackbaud:gradle-internal-test:4.+') {
+    sharedTestCompile gradleTestKit()
+    sharedTestCompile ("com.blackbaud:gradle-internal-test:${gradleInternalMajorVersion}") {
         exclude group: 'org.spockframework'
     }
-    testCompile('org.spockframework:spock-core:1.0-groovy-2.4') {
+    sharedTestCompile('org.spockframework:spock-core:1.2-groovy-2.4') {
```
Comment (Blackbaud-BethRichardson) [inline build.gradle]: why dont you add the dependencies plugin here?

File: src/componentTest/resources/expectedTemplateContent/deployable/build.gradle [hunk 2]
```diff
@@ -20,35 +19,28 @@ allprojects {
...
     }
 
     dependencies {
-        compileOnly "org.projectlombok:lombok:${lombokVersion}"
+        compileOnly "org.projectlombok:lombok"
 
-        mainTestCompile "com.blackbaud:common-test:4.+"
-        mainTestCompileOnly "org.projectlombok:lombok:${lombokVersion}"
-        mainTestCompileOnly "org.codehaus.groovy:groovy-all:${groovyVersion}"
+        mainTestCompileOnly "com.blackbaud:common-test:4.+"
```
Comment (Blackbaud-BethRichardson) [inline src/componentTest/resources/expectedTemplateContent/deployable/build.gradle:29]: consider pinning common-test as well in dependencies plugin

File: src/componentTest/resources/expectedTemplateContent/integration-test/build.gradle [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
 }
 
 dependencies {
-    integrationTestCompile "com.google.guava:guava:23.6.1-jre"
-    integrationTestCompile "com.blackbaud:common-spring-boot-rest-test:${springBootVersion}-3.+"
+    integrationTestCompile "com.google.guava:guava:27.0.1-jre"
+    integrationTestCompile "com.blackbaud:common-deployable-spring-boot-rest-test:${springBootVersion}-0.+"
     integrationTestCompile "org.springframework.boot:spring-boot-starter-test:${springBootVersion}"
-    integrationTestCompile "com.blackbaud:common-geb:3.+" 
+
```
Comment (Blackbaud-BethRichardson) [inline src/componentTest/resources/expectedTemplateContent/integration-test/build.gradle]: what happened to geb?

File: src/main/groovy/com/blackbaud/templates/project/DatasourceProject.groovy [hunk 4]
```diff
@@ -44,7 +44,7 @@ class DatasourceProject {
         applyPostgresCompileDependencies()
         applyPostgresApplicationProperties()
         applyTestCleanupSql()
-        basicProject.appendServiceToAppDescriptor("postgres-shared")
+//        basicProject.appendServiceToAppDescriptor("postgres-shared")
```
Comment (Blackbaud-BethRichardson) [inline src/main/groovy/com/blackbaud/templates/project/DatasourceProject.groovy:47]: why include this commented?

File: src/main/groovy/com/blackbaud/templates/project/KafkaProject.groovy [hunk 5]
```diff
@@ -33,7 +33,7 @@ class KafkaProject {
         buildFile.appendAfterLine(/sharedTestCompile/,
                 '    sharedTestCompile "com.blackbaud:common-kafka-test:${commonKafkaVersion}"')
 
-        basicProject.appendServiceToAppDescriptor("kafka")
+//        basicProject.appendServiceToAppDescriptor("kafka")
```
Comment (Blackbaud-BethRichardson) [inline src/main/groovy/com/blackbaud/templates/project/KafkaProject.groovy:36]: why leave commented?

File: src/main/groovy/com/blackbaud/templates/project/ProjectFile.groovy [hunk 6]
```diff
@@ -79,6 +102,24 @@ import ${importToAdd}${eol}""")
...
+        List<String> lines = readLines()
+        int index = indexOf(lines, match)
+
+        if (index >= 0) {
+            for (int i = index + 1; i < lines.size(); i++) {
+                if (lines[i] =~ /${match}/) {
+                    index = i
+                }
+                if (index != i) {
+                    lines.addAll(i, lineToAdd)
```
Comment (Blackbaud-BethRichardson) [inline src/main/groovy/com/blackbaud/templates/project/ProjectFile.groovy]: should this be add?

**Review by Blackbaud-BethRichardson — CHANGES_REQUESTED** [review]

File: src/main/groovy/com/blackbaud/templates/project/DatasourceProject.groovy (hunk 4 above)
Comment (Blackbaud-MikeLueders) [reply src/main/groovy/com/blackbaud/templates/project/DatasourceProject.groovy:47]: well, the appendServiceToAppDescriptor method got removed but i haven't yet removed the postgres code generation which i'm going to do in a follow-up PR, was going to remove the line then

File: build.gradle (hunk 1 above)
Comment (Blackbaud-MikeLueders) [reply build.gradle]: done

File: src/componentTest/resources/expectedTemplateContent/deployable/build.gradle (hunk 2 above)
Comment (Blackbaud-MikeLueders) [reply src/componentTest/resources/expectedTemplateContent/deployable/build.gradle:29]: https://github.com/blackbaud/gradle-internal/pull/268

File: src/componentTest/resources/expectedTemplateContent/integration-test/build.gradle (hunk 3 above)
Comment (Blackbaud-MikeLueders) [reply src/componentTest/resources/expectedTemplateContent/integration-test/build.gradle]: done

File: src/main/groovy/com/blackbaud/templates/project/ProjectFile.groovy (hunk 6 above)
Comment (Blackbaud-MikeLueders) [reply src/main/groovy/com/blackbaud/templates/project/ProjectFile.groovy]: done

**Review by Blackbaud-BethRichardson — APPROVED** [review]

**Pull request merged**